explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
+## Usage examples
 
-Non-goals are:
+The simplest possible usage of the API is a single call to `Badge.set` from a
+foreground context (which might be used to show an unread count in an email
+app):
 
-* To provide an arbitrary image badge. The  web platform already provides this capability via favicons.
+```js
+Badge.set(getUnreadCount());
+```
 
-Possible areas for expansion:
+This will set the badge for all pages and apps in the current origin until it is
+changed. If `getUnreadCount()` (the argument to `Badge.set`) is 0, it will
+automatically clear the badge.
 
-* Support rendering a small status indicator (e.g., a music app shows ▶️ or ⏸️; a weather app shows ⛈️ or ⛅️).
-* Support setting the badge from a service worker (e.g. an email app updating an unread count in the background).
-* Support badges on bookmark icons (this is probably up to user agents to support, and may not need anything new API wise).
-* Support for non-inherited badges (see [Issue 42](https://github.com/WICG/badging/issues/42), e.g. a page which normally shows a status might not want to fall back to the notification count badged on the origin when the status is cleared).
-* Support for unicode glyph badges (e.g. 😀, 😍, ❤, to render a more diverse range of statuses).
+If you just want to show a status indicator flag without a number, use the
+Boolean mode of the API by calling `Badge.set` without an argument, and
+`Badge.clear` (which might be done to indicate that it is the player's turn to
+move in a multiplayer game):
 
-Examples of sites that may use this API:
+```js
+if (myTurn())
+  Badge.set();
+else
+  Badge.clear();
+```
 
-* Chat, email, and social apps could signal that new messages have arrived.
-* Any application that needs to signal that user action is required (e.g., in a turn-based game, when it is the player's turn).
+The reason we are considering both the "document" and "handle" contexts in the
+same API, and not two separate badging APIs, is so that in the common case,
+developers can just set a badge for the origin and have it show up in whatever
+places the user agent wants to show it. However, if you just want to badge a
+specific set of URLs, and not the whole site, use the `scope` option (which
+might be done if you have a different "unread count" on each page):
 
-Advantages of using the badging API over notifications:
+```js
+Badge.set(getUnreadCount(location.pathname), {scope: location});
+```
 
-* Can be used for much higher frequency events than notifications, because each new event does not disrupt the user.
-* There may be no need to request permission to use the badging API, since it is much less invasive than a notification.
+The scope is a URL prefix; the badge is applied on all pages whose URL [starts
+with](https://www.w3.org/TR/appmanifest/#dfn-within-scope) that prefix.
```

Comment:
Maybe... it doesn't really matter. I prefer the app definition (because I wrote it -- ha!) but mostly because it's got notes that explain why it does a prefix match instead of the obvious one.

Reply:
No strong feelings here.